CMakeLists.txt

```diff
@@ -67,9 +67,14 @@ check_include_file (execinfo.h HAVE_EXECINFO_H)
 check_include_file (ucontext.h HAVE_UCONTEXT_H)
 
 # check functions
+
+# set libraries to be used with check_function_exists
+set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} gc)
```

Comment:
Please use `${LIBGC_LIBRARIES}` to account for non-standard installations. The find_package command will return the library with the right path.
Also, please reset `CMAKE_REQUIRED_LIBRARIES` after the check. You do not necessarily want libgc to be linked to all targets.